File: Classes/Models/PDFRCSectionInfoAdapter.m
//
//  PDFRCSectionInfoAdapter.m
//  PDStrategy
//
//  Created by Pavel Deminov on 22/10/2017.
//

#import "PDFRCSectionInfoAdapter.h"

@implementation PDFRCSectionInfoAdapter

+ (instancetype)adapterWith:(id <NSFetchedResultsSectionInfo>)sectionInfo sectionIdentifier:(NSString *)sectionIdentifier {
    PDFRCSectionInfoAdapter *adapter = [[PDFRCSectionInfoAdapter alloc] initWithFRCSectionInfo:sectionInfo sectionIdentifier:sectionIdentifier];
    return adapter;
}

- (instancetype)initWithFRCSectionInfo:(id <NSFetchedResultsSectionInfo>)sectionInfo sectionIdentifier:(NSString *)sectionIdentifier {
    self = [super init];
    if (self) {
        self.frcSectionInfo = sectionInfo;
        self.sectionIdentifier = sectionIdentifier;
    }
    return self;
}

- (NSArray *)items {
    return self.frcSectionInfo.objects;
}

- (NSString *)title {
    return self.frcSectionInfo.name;
}

@end

File: Example/PDStrategy/PDStrategy/PDStrategy/Adapters/PDFRCSectionInfoAdapter.swift
//
//  PDFRCSectionInfoAdapter.swift
//  PDStrategy
//
//  Created by Pavel Deminov on 08/11/2017.
//  Copyright © 2017 Pavel Deminov. All rights reserved.
//

import UIKit
import CoreData

class PDFRCSectionInfoAdapter: NSObject, PDSectionInfo {
    var frcSectionInfo: NSFetchedResultsSectionInfo?
    var title: String? {
        get {
            return frcSectionInfo?.name
        }
    }
    var items: [PDItemInfo]? {
        get {
            return frcSectionInfo?.objects as? [PDItemInfo]
        }
    }
    
    init(frcSectionInfo sectionInfo: NSFetchedResultsSectionInfo) {
        super.init()
        self.frcSectionInfo = sectionInfo
        setup()
    }
    
    func setup() {
        
    }
    
}
